lib/avatarDialog/avatar_dialog.dart

```diff
@@ -0,0 +1,165 @@
+  int lastSelectedIndex = 19;
+
+  Future _scrollToIndex(int index) async {
+    await controller.scrollToIndex(index,
+        preferPosition: AutoScrollPosition.begin);
+    setState(() {
+      selectedIndex = index - 1;
+    });
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    _scrollToIndex(lastSelectedIndex);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Dialog(
+      backgroundColor: const Color(0xFFFFFFFF),
+      alignment: Alignment.bottomCenter,
+      insetPadding: const EdgeInsets.fromLTRB(10.0, 10.0, 10.0, 10.0),
+      shape: const RoundedRectangleBorder(
+        borderRadius: BorderRadius.all(Radius.circular(35.0)),
+      ),
+      child: SizedBox(
+        height: MediaQuery.of(context).size.height * 0.65,
+        width: MediaQuery.of(context).size.width,
+        child: Column(
+          crossAxisAlignment: CrossAxisAlignment.center,
+          children: <Widget>[
+            Container(
+              margin: const EdgeInsets.only(top: 44.0),
+              child: const AutoSizeText(
+                "Select Avatar",
+                style: TextStyle(
+                  color: Color(0xFF1353CB),
+                  fontSize: 26.0,
+                  fontWeight: FontWeight.bold,
+                  fontFamily: 'Poppins',
+                ),
+                textAlign: TextAlign.center,
+                maxLines: 1,
+              ),
+            ),
+            Expanded(
+              child: Container(
+                margin: const EdgeInsets.fromLTRB(15.0, 22.0, 15.0, 25.0),
+                child: GridView.builder(
+                    physics: const ScrollPhysics(),
+                    controller: controller,
+                    gridDelegate:
+                        const SliverGridDelegateWithFixedCrossAxisCount(
+                      crossAxisCount: 3,
+                      mainAxisSpacing: 3,
+                      crossAxisSpacing: 3,
+                    ),
+                    itemCount: listLength,
+                    itemBuilder: (context, index) {
+                      bool checked = (index == selectedIndex);
+                      return AutoScrollTag(
+                        controller: controller,
+                        index: index,
+                        key: ValueKey(index),
+                        child: InkWell(
+                          onTap: () {
+                            setState(() {
+                              selectedIndex = index;
+                            });
+                          },
+                          child: SizedBox(
+                            height: MediaQuery.of(context).size.width * 0.3,
+                            width: MediaQuery.of(context).size.width * 0.3,
+                            child: Stack(children: <Widget>[
+                              Card(
+                                margin: const EdgeInsets.fromLTRB(
+                                    7.0, 2.0, 7.0, 3.0),
```

Comment:
Why is there a margin here, grid view spacing should be handled in gridDelegate

Reply:
removed and added spacing in gridDelegate